server/src/recipe/handler/recipe.handler.ts

```diff
@@ -1,150 +1,122 @@
 
 import {
-  ApiError, DataType, ERRORS, ImageService, SocoboRequest
+  ApiError, DataType, DbError, ERRORS, ImageService, SocoboRequest
 } from "../../app/index";
-
-import { Recipe, RecipeMiddleware } from "../index";
+import { DbExtension } from "../../db/interface/db-extension";
+import { Recipe} from "../index";
 
 export class RecipeHandler {
 
-  private _db: any; // TODO: any wird zu DbExtension //IDatabase<DbExtensions>&DbExtensions;
-  private _recipeMiddleware: RecipeMiddleware;
-  private _imgService: ImageService;
+  constructor (private _db: DbExtension, private _imgService: ImageService) {}
 
-  constructor (db: any, recipeMiddleware: RecipeMiddleware, imgService: ImageService) {
-    this._db = db;
-    this._recipeMiddleware = recipeMiddleware;
-    this._imgService = imgService;
+  private _sendError = (res: Response) => {
+    return (error: any) => {
+      res.status(error.statusCode).json(error.forResponse());
+    };
   }
 
-  public getById = (req: Request, res: Response): void => {
-    this._db.recipes.getById(req.params.id)
-      .then((result: Recipe) => res.status(200).json(result))
-      .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
+  public getById = async (req: Request, res: Response) => {
+    try {
+      const result = await this._db.recipe.getById(req.params.id);
+      res.status(200).json(result);
+    } catch (error) {
+      this._sendError(res)(error);
+    }
   }
 
-  public getAll = (req: Request, res: Response): void => {
-
-    const queryPramas = req.query;
-    const allQueryParams = Object.getOwnPropertyNames(queryPramas);
-    const firstQueryParam = allQueryParams[0];
-    const valueFirstQueryParam = queryPramas[firstQueryParam];
-
-    if (firstQueryParam && valueFirstQueryParam) {
-      if (!(new Recipe().fields.has(firstQueryParam))) {
-        const e = new ApiError(ERRORS.RECIPE_INVALID_FIELD.withArgs(firstQueryParam))
-          .addSource(RecipeHandler.name)
-          .addSourceMethod("getAll()");
-        res.status(e.statusCode).json(e.forResponse());
-      }
-      this._db.recipes.getByField(firstQueryParam, valueFirstQueryParam)
-        .then((result: Recipe[]) => res.status(200).json(result))
-        .catch((e: any) => res.status(e.statusCode).json(e.forResponse()));
-    } else {
-      this._db.recipes.getAll()
-        .then((result: Recipe[]) => res.status(200).json(result))
-        .catch((e: any) => {
-          if (e.code === ERRORS.RECIPE_NON_AVAILABLE.code) {
-            res.status(200).json([]);
-          }
-          res.status(e.statusCode).json(e.forResponse());
-      });
+  public getAll = async (req: Request, res: Response) => {
+    try {
+      const result = await this._db.recipe.getAll();
+      res.status(200).json(result);
+    } catch (error) {
+      this._sendError(res)(error);
     }
   }
 
-  public searchByField = (req: Request, res: Response): void => {
+  public searchByField = async (req: Request, res: Response) => {
```

Comment:
should we also add searchByFields?

Reply:
Recipe already had this feature. We don't need to implement this in each domain but I think it makes sense to search in recipes. What do u think?

Reply:
> Recipe already had this feature. We don't need to implement this in each domain
i can't find it? what did you mean?

> I think it makes sense to search in recipes. What do u think?
yes i think this makes sense.